packages/remix/src/utils/web-fetch.ts

```diff
@@ -147,8 +147,8 @@ export const normalizeRemixRequest = (request: RemixRequest): Record<string, any
     query: parsedURL.query,
     href: parsedURL.href,
     method: request.method,
-    // @ts-expect-error - not sure what this supposed to do
-    headers: headers[Symbol.for('nodejs.util.inspect.custom')](),
+    // Remix now uses undici, so use standard way to create headers object
+    headers: Object.fromEntries(headers),
```

Comment:
Could you add a test case for this?

Reply:
Sure. Also CI failed so it looks like you support older JS versions. I'll come up with an alternative.

Reply:
I'm having an issue writing tests. It looks like `jest` isn't using the actual `Request` and `Headers` implementation, so these are _mocked_ out (see `getIpAddress.test.ts` for example).

So, I mocked out a basic `Request/Headers` for this test. 

I wrote a helper to convert `Headers` to object since you support older JavaScript without `Object.fromEntries()`

It's crazy that I had to jump through a bunch of hoops to not only get it to work but also satisfy TypeScript and your lint config.

So here's my initial version:
```ts
function objectFromHeaders(headers: Headers): Record<string, string> {
  const result: Record<string, string> = {};
  for (const [key, value] of headers.entries()) {
    result[key] = value;
  }
  return result;
}
```
This results in a test failure because the mocked `Headers` doesn't have `entries()` even though it's defined in the type.
![image](https://github.com/getsentry/sentry-javascript/assets/47168/643287a0-9989-4836-942c-ebfdc3238b57)

![image](https://github.com/getsentry/sentry-javascript/assets/47168/b365a34f-07a2-4530-bda7-113c1461e787)

So then I tried removing `entries()`
```ts
function objectFromHeaders(headers: Headers): Record<string, string> {
  const result: Record<string, string> = {};
  for (const [key, value] of headers) {
    result[key] = value;
  }
  return result;
}
```
Then I got this error. Again, because `Headers` is not a true implementation
![image](https://github.com/getsentry/sentry-javascript/assets/47168/e0d66780-82f3-4715-b084-73a053432cc5)

Funny enough, if I go back to using `entries()`, it passes the test as long as I add `// @ts-ignore`. Apparently, the previous error was a TypeScript error and not a runtime one.
```ts
function objectFromHeaders(headers: Headers): Record<string, string> {
  const result: Record<string, string> = {};
  // @ts-ignore
  for (const [key, value] of headers.entries()) {
    result[key] = value;
  }
  return result;
}
```
![image](https://github.com/getsentry/sentry-javascript/assets/47168/4db7664b-e15b-47cb-bf58-638ec68daa27)
But unfortunately, it won't pass lint
![image](https://github.com/getsentry/sentry-javascript/assets/47168/45bc8bfe-931f-4de6-ab2b-77b1e1804d8b)

So I had to write all this (thanks ChatGPT) just to get tests and lint to pass 😔

```ts
function objectFromHeaders(headers: Headers): Record<string, string> {
  const result: Record<string, string> = {};
  let iterator: IterableIterator<[string, string]>;

  if (hasIterator(headers)) {
    iterator = getIterator(headers) as IterableIterator<[string, string]>;
  } else {
    return result;
  }

  for (const [key, value] of iterator) {
    result[key] = value;
  }
  return result;
}

type IterableType<T> = {
  [Symbol.iterator]: () => Iterator<T>;
};

function hasIterator<T>(obj: T): obj is T & IterableType<unknown> {
  return obj !== null && typeof (obj as IterableType<unknown>)[Symbol.iterator] === 'function';
}

function getIterator<T>(obj: T): Iterator<unknown> {
  if (hasIterator(obj)) {
    return (obj as IterableType<unknown>)[Symbol.iterator]();
  }
  throw new Error('Object does not have an iterator');
}
```
![image](https://github.com/getsentry/sentry-javascript/assets/47168/3c159ce1-5941-44a6-89bd-49fc8db0f0a0)

Yay, I think.

Let me know if I should continue with the PR.